EmployeeController.java

```diff
@@ -29,38 +35,43 @@ public class EmployeeController {
     private EmployeeRepository employeeData;
 
     @RequestMapping(value = "/addNewEmployee", method = RequestMethod.POST)
-    public Employee addEmployee(@RequestBody Employee employee) {
+    public ResponseEntity<Object> addEmployee(@RequestBody Employee employee) {

```

Comment:
You could have validated the request using spring validator framework

Reply:
Agreed. In the initial stage we used Validation framework, but later architects and clients decided to use custom validations because of overhead by the lot of annotations. When annotations got increased, its hard to debug and many developers felt uncomfortable dealing with it.

And here I used custom validations to make sure I am aware of regex which needed many times. Thanks